### PR TITLE
[MME][mconfig] Update lte mconfig to use configured mme pool configs if they exist

### PR DESCRIPTION
**Pull request by mpgermano**
Signed-off-by: Michael Germano <mgermano@fb.com>

## Summary

Recent PR https://github.com/magma/magma/pull/3857 added support for gateway pools within an LTE
network. Here we update the mconfig for an LteGateway to use these configs if they exist.

## Test Plan

Tested locally between orc8r and lte gateway to ensure mconfigs
were properly set.

Added unit tests.

### CODE REVIEW

File: lte/cloud/go/services/lte/servicers/builder_servicer.go [hunk 1]
```diff
@@ -301,6 +307,31 @@ func getHEConfig(gwConfig *lte_models.GatewayHeConfig) *lte_mconfig.PipelineD_HE
...
+
+func getMMEPoolConfigs(networkID string, poolingConfig lte_models.CellularGatewayPoolRecords, cellGateway configurator.NetworkEntity, graph configurator.EntityGraph) (*lte_models.CellularGatewayPoolRecord, uint32, error){
+	// Currently, having multiple (mme group ID, mme code, mme relative
+	// capacity) tuples is unsupported. As such, use the first pool record
+	// to set all of these values.
+	if len(poolingConfig) == 0 {
+		return &lte_models.CellularGatewayPoolRecord{
+			MmeCode:             1,
+			MmeRelativeCapacity: 10,
+		}, 1, nil
```
Comment (ulaskozat) [inline lte/cloud/go/services/lte/servicers/builder_servicer.go]: not obvious unless someone reads the whole function implementation what the second returned value of uin32 signify.

File: lte/cloud/go/services/lte/servicers/builder_servicer.go [hunk 2]
```diff
@@ -301,6 +307,31 @@ func getHEConfig(gwConfig *lte_models.GatewayHeConfig) *lte_mconfig.PipelineD_HE
 		EncryptionKey:          gwConfig.EncryptionKey,
 	}
 }
+
+func getMMEPoolConfigs(networkID string, poolingConfig lte_models.CellularGatewayPoolRecords, cellGateway configurator.NetworkEntity, graph configurator.EntityGraph) (*lte_models.CellularGatewayPoolRecord, uint32, error){
```
Comment (ulaskozat) [inline lte/cloud/go/services/lte/servicers/builder_servicer.go]: putting a comment before the function to specify the default return values would be nice.

Comment (hcgatewood) [reply lte/cloud/go/services/lte/servicers/builder_servicer.go]: +1. Alternative is named returns. I doc comment except when named returns also serve alternative purpose

File: lte/cloud/go/services/lte/servicers/builder_servicer.go (hunk 1 above)
Comment (hcgatewood) [reply lte/cloud/go/services/lte/servicers/builder_servicer.go]: +1. These 1/0 returns are magic numbers https://flaviocopes.com/magic-numbers/

File: lte/cloud/go/services/lte/servicers/builder_servicer.go [hunk 3]
```diff
@@ -301,6 +307,31 @@ func getHEConfig(gwConfig *lte_models.GatewayHeConfig) *lte_mconfig.PipelineD_HE
...
+		}, 1, nil
+	}
+	pool, err := graph.GetFirstAncestorOfType(cellGateway, lte.CellularGatewayPoolEntityType)
+	if err != nil {
+		return nil, 0, err
+	}
+	poolRecord := poolingConfig[0]
+	cfg, ok := pool.Config.(*lte_models.CellularGatewayPoolConfigs)
+	if !ok {
+		err := fmt.Errorf("Unable to convert gateway pool config for pool '%s'; pool has invalid config", pool.Key)
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/servicers/builder_servicer.go]: Lowercase error strings https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: lte/cloud/go/services/lte/servicers/builder_servicer.go [hunk 4]
```diff
@@ -301,6 +307,31 @@ func getHEConfig(gwConfig *lte_models.GatewayHeConfig) *lte_mconfig.PipelineD_HE
...
+		return nil, 0, err
+	}
+	mmeGroupID := cfg.MmeGroupID
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/servicers/builder_servicer.go]: This line feels superfluous

File: lte/cloud/go/services/lte/servicers/builder_servicer_test.go [hunk 5]
```diff
@@ -831,6 +836,134 @@ func TestBuilder_BuildUnmanagedEnbConfig(t *testing.T) {
...
+		Type: lte.CellularGatewayPoolEntityType, Key: "pool1",
+		Config: &lte_models.CellularGatewayPoolConfigs{
+			MmeGroupID: 2,
+		},
+	}
+	lteGatewayConfigs := newDefaultGatewayConfig()
+	lteGatewayConfigs.Pooling = lte_models.CellularGatewayPoolRecords{
+		{
+			GatewayPoolID: "pool1",
+			MmeCode: 1,
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/servicers/builder_servicer_test.go]: Can you make this MmeCode value != 1, just to have one test case where the value isn't =1

**Review by hcgatewood — APPROVED** [review]
Lgtm, please just fix my+Ulas's couple small comments before merging

**Pull request merged**